Source/MaestroAdvection.cpp

```diff
@@ -112,26 +117,119 @@ Maestro::MakeUtrans (const Vector<MultiFab>& utilde,
             // use macros in AMReX_ArrayLim.H to pass in each FAB's data,
             // lo/hi coordinates (including ghost cells), and/or the # of components
             // We will also pass "validBox", which specifies the "valid" region.
-#if (AMREX_SPACEDIM == 2)
-            mkutrans_2d(
-#elif (AMREX_SPACEDIM == 3)
+
             mkutrans_3d(
-#endif
-                        &lev, domainBox.loVect(), domainBox.hiVect(),
-                        tileBox.loVect(), tileBox.hiVect(),
+                        &lev, AMREX_ARLIM_ANYD(domainBox.loVect()), AMREX_ARLIM_ANYD(domainBox.hiVect()),
+                        AMREX_ARLIM_ANYD(tileBox.loVect()), AMREX_ARLIM_ANYD(tileBox.hiVect()),
                         BL_TO_FORTRAN_FAB(utilde_mf[mfi]), utilde_mf.nGrow(),
                         BL_TO_FORTRAN_FAB(ufull_mf[mfi]), ufull_mf.nGrow(),
                         BL_TO_FORTRAN_3D(utrans_mf[mfi]),
                         BL_TO_FORTRAN_3D(vtrans_mf[mfi]),
-#if (AMREX_SPACEDIM == 3)
+                        BL_TO_FORTRAN_3D(vtrans_mf[mfi]),
                         BL_TO_FORTRAN_3D(wtrans_mf[mfi]),
                         BL_TO_FORTRAN_3D(w0macx_mf[mfi]),
                         BL_TO_FORTRAN_3D(w0macy_mf[mfi]),
                         BL_TO_FORTRAN_3D(w0macz_mf[mfi]),
-#endif
                         w0.dataPtr(), dx, &dt, bcs_u[0].data(), phys_bc.dataPtr());
 
         } // end MFIter loop
+
+#else
+// AMREX_SPACEDIM == 2
+
+#ifdef AMREX_USE_CUDA
+        int* bc_f = prepare_bc(bcs_u[0].data(), 1);
+        set_bc_launch_config();
```

Comment:
We don't need to do `set_bc_launch_config()` here (it's only necessary for correctness for the AMReX BC routines that do a custom threadblock synchronization step internally) and actually we don't want to call it because it's a lower performance configuration.

Reply:
I got rid of these in the next PR